lib/s3middleware/objectRestore.ts

```diff
@@ -0,0 +1,141 @@
+    // RestoreRequest.Days must be greater than or equal to 1
+    const daysValue = Number.parseInt(days[0], 10);
+    if (Number.isNaN(daysValue)) {
+        const desc = `RestoreRequest.Days is invalid type. [${days}]`;
+        const error = errors.MalformedXML.customizeDescription(desc);
+        return { error };
+    }
+    if (daysValue < 1) {
+        const desc = `RestoreRequest.Days must be greater than 0. [${days}]`;
+        const error = errors.MalformedXML.customizeDescription(desc);
+        return { error };
+    }
+    if (daysValue > 2147483647) {
+        const desc = `RestoreRequest.Days must be less than 2147483648. [${days}]`;
+        const error = errors.MalformedXML.customizeDescription(desc);
+        return { error };
+    }
+    return { days: daysValue }
+}
+
+/**
+ * validateTier - validate retain until date
+ * @param tier - parsed tier string
+ * @return - contains retain until date or error
+ */
+function validateTier(tier?: string) {
+    // If Tier is specified, must be "Expedited" or "Standard" or "Bulk"
+    const tierList = ['Expedited', 'Standard', 'Bulk'];
+    if (tier && tier[0] && !tierList.includes(tier[0])) {
+        const desc = `RestoreRequest.Tier is invalid value. [${tier}]`;
+        const error = errors.MalformedXML.customizeDescription(desc);
+        return { error };
+    }
+    // If do not specify, set "Standard"
+    return { tier: tier && tier[0] ? tier[0] : 'Standard' };
+}
+
+/**
+ * validate restore request xml
+ * @param parsedXml - parsed restore request xml object
+ * @return - contains restore request information on success,
+ * error on failure
+ */
+function validateRestoreRequestParameters(parsedXml?: any) {
+    if (!parsedXml) {
+        const desc = 'request xml is undefined or empty';
+        const error = errors.MalformedXML.customizeDescription(desc);
+        return { error };
+    }
+    const restoreRequest = parsedXml.RestoreRequest;
+    if (!restoreRequest) {
+        const desc = 'request xml does not contain RestoreRequest';
+        const error = errors.MalformedXML.customizeDescription(desc);
+        return { error };
+    }
+    const daysObj = validateDays(restoreRequest.Days);
+    if (daysObj.error) {
+        return { error: daysObj.error };
+    }
+    const tierObj = validateTier(restoreRequest.Tier);
+    if (tierObj.error) {
+        return { error: tierObj.error };
+    }
+    return { days: daysObj.days, tier: tierObj.tier };
+}
+
+/**
+ * parseRestoreRequestXml - Parse and validate xml body, returning callback with
+ * object restoreReqObj: { days: <value>, tier: <value> }
+ * @param xml - xml body to parse and validate
+ * @param log - Werelogs logger
+ * @param cb - callback to server
+ * @return - calls callback with object restore request or error
+ */
+export function parseRestoreRequestXml(
+    xml: string,
+    log: werelogs.Logger,
+    cb: (err: ArsenalError | null, data?: any) => void,
+) {
+    parseString(xml, (err, result) => {
```

Comment:
I would use async await here instead of a callback

Reply:
yes, use parseStringPromise instead